lib/src/components/NxNumberInput/types.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import * as PropTypes from 'prop-types';
+import { FormEvent, InputHTMLAttributes, ReactNode } from 'react';
+
+export type ValidationErrors = string | string[] | null;
+export type Validator = ((val: string) => ValidationErrors) | null;
+
+/**
+ * Props whose value will typically change as the text input on the page is used, as opposed to props which
+ * are typically constant for a given text input.
+ */
+export interface StateProps {
+  value: string;
+  trimmedValue: string;
+  isPristine: boolean;
+  validationErrors?: ValidationErrors;
+}
+
+// Imported props from react to provide support for all the regular html attributes
+type FusedProps = InputHTMLAttributes<HTMLInputElement>;
+// Leave out props to be re-defined
+export type HTMLProps = Omit<FusedProps, 'onChange'|'onKeyPress'|'defaultValue'>;
+
+export type NumberInputElement = HTMLInputElement;
```

Comment:
This isn't useful.  It made sense in `NxTextInput` because it's a union of two types there.  Here it's just an alias

Reply:
> This isn't useful. It made sense in `NxTextInput` because it's a union of two types there. Here it's just an alias

Right: https://github.com/sonatype/sonatype-react-shared-components/pull/486/commits/2551aa67d5324bfe097d4fe273a5d47ef5e235fb